extensions-core/kafka-indexing-service/src/main/java/io/druid/indexing/kafka/IncrementalPublishingKafkaIndexTaskRunner.java

```diff
@@ -429,9 +429,6 @@ public void run()
           // if stop is requested or task's end offset is set by call to setEndOffsets method with finish set to true
           if (stopRequested.get() || sequences.get(sequences.size() - 1).isCheckpointed()) {
```

Comment:
I discussed with @jihoonson because I noticed that previously it was setting the state to publishing and then going through a bunch of logic not related to publishing if stop wasn't user requested, and this is what was intended to happen here. As I understand, currently it will harmlessly but needlessly fall through the logic below before breaking out and publishing.